apps/io-profile-async/src/repositories/__tests__/profile-email.test.ts

```diff
@@ -0,0 +1,102 @@
+import { beforeEach, describe, vi, it, expect } from "vitest";
+import * as TE from "fp-ts/lib/TaskEither";
+import * as O from "fp-ts/lib/Option";
+import * as E from "fp-ts/lib/Either";
+import {
+  IProfileEmailReader,
+  IProfileEmailWriter,
+  ProfileEmail,
+  ProfileEmailWriterError
+} from "@pagopa/io-functions-commons/dist/src/utils/unique_email_enforcement";
+import { EmailString } from "@pagopa/ts-commons/lib/strings";
+import { mockProfiles } from "../../functions/__mocks__/profile-repository.mock";
+import { ProfileEmailRepository } from "../profile-emails";
+
+const aProfileEmail: ProfileEmail = {
+  fiscalCode: mockProfiles[0].fiscalCode,
+  email: mockProfiles[0].email as NonNullable<EmailString>
+};
+
+const dataTableInsertMock = vi.fn().mockResolvedValue(void 0);
+const dataTableDeleteMock = vi.fn().mockResolvedValue(void 0);
+
+const dataTableMock = ({
+  insert: dataTableInsertMock,
+  delete: dataTableDeleteMock
+} as unknown) as IProfileEmailWriter & IProfileEmailReader;
+
+describe.each`
+  kind             | isInsert
+  ${"emailInsert"} | ${true}
+  ${"emailDelete"} | ${false}
```

Comment:
Even though this trick reduces the code, it makes it harder to read. Can we split them instead?

Reply:
split present in https://github.com/pagopa/io-auth-n-identity-domain/pull/224/commits/02e4e7ab0270136c7da543c52da54afb4ac6ae19